smart_open/ftp.py

```diff
@@ -169,4 +169,5 @@ def full_close(self):
     fobj.socket = socket
     fobj.conn = conn
     fobj.close = types.MethodType(full_close, fobj)
+    # fobj.name = path
```

Comment:
If we don't need it, then let's remove it instead of commenting

Reply:
yep, the call was hanging because `name` is a property of a socket file: https://github.com/python/cpython/blob/v3.11.1/Lib/socket.py#L758

so this PR is now adding tests and an assertion, meaning the tests are not able to reproduce the issue in question.

Reply:
`unable to transparently decompress <_io.BufferedReader name=580> because it seems to lack a string-like .name`

name=580, returned by fileno() in the link above, is not reproducted in integration tests